### ISSUE TITLE
Wrong line gets copied over when selecting text

### ISSUE THREAD

**Issue by ltguillaume** [WinMerge/winmerge#2499]
Two files:

**A (left)**
```
Keyboard Layout
===============

Key                    Action
~~~                    ~~~~~~

```
**B (right)**
```
Keyboard Layout
===============

aKey                    Action
a~~~                    ~~~~~~
a
```
Now, if I select the line `Key                    Action` in file A by putting the cursor before the `K`, then hold `Shift` and press `Arrow Down`, I would expect that a copy via `Alt` `Right` would result in the line `aKey                    Action` to become `Key                    Action` on the right.

Instead, the line `a~~~                    ~~~~~~` becomes `~~~                    ~~~~~~`.

In this situation, the cursor might be in front of the line `~~~                    ~~~~~~`, but it doesn't make sense that the line the cursor is in front of is now copied over instead of the selected text.

WinMerge 2.16.44
Windows 10

**Comment by sdottaka** [comment]
I don't yet know why it works this way, but if you want to copy lines selected with Alt+Right, select "Line" for the "Copy granularity for selected differences" option in the Editor→Compare/Merge category of the Options window.

![image](https://github.com/user-attachments/assets/2218c9a1-822d-4716-aae2-d8dfcb62cd65)

**Comment by sdottaka** [comment]
This issue has been resolved with commit c73bd25. Thank you for your report.
